### PR TITLE
[botkit] Add Slack Adapter Bot sample 

### PR DESCRIPTION
**Pull request by LCarbonellRod**
### Description

- Add new sample. Echo bot to test the Slack Adapter.

- A README file is added with the steps to connect the bot with Slack Api to use the bot.

_**Note**: This PR will fail until the Slack Adapter nuget package gets published._

### CODE REVIEW

File: samples/csharp_dotnetcore/53.SlackAdapterBot/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:51709/",
```
Comment (matiasroldan6) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/Properties/launchSettings.json]: should this be set to port 3978 like the rest of the adapters?
Same below

**Review by Aliandi — APPROVED** [review]

File: samples/csharp_dotnetcore/53.SlackAdapterBot/Bots/EchoBot.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using SlackAPI;
+using Attachment = Microsoft.Bot.Schema.Attachment;
+
+namespace Microsoft.Bot.Builder.Adapters.Slack.TestBot.Bots
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/Bots/EchoBot.cs]: Please change all namespaces to SlackAdapterBot

File: samples/csharp_dotnetcore/53.SlackAdapterBot/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+﻿# EchoBot using Slack Adapter
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/README.md]: Please change the title to `Slack Adapter ` to keep consistency in all samples.

File: samples/csharp_dotnetcore/53.SlackAdapterBot/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+- [.NET Core SDK](https://dotnet.microsoft.com/download) version 2.1
+
+  ```bash
+  # determine dotnet version
+  dotnet --version
+  ```
+
+## To try this sample
+
+1 - Clone the repository
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/README.md]: Remove the step number in these 6 steps so the snippets get shown correctly.

File: samples/csharp_dotnetcore/53.SlackAdapterBot/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+### Connect the bot with Slack
+
+1 - Create a SlackApp following this [guide](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-connect-slack?view=azure-bot-service-4.0)
+
+2 - Copy the _Verification Token_, and the _Signing Secret_ from the **Basic Information** tab and the _Bot User OAuth Access Token_ from the **Oauth & Permissions** tab (after installing the app in a Workspace). These credentials will be needed to connect the bot to Slack.
+
+Some things to consider:
+
+a. In the **Oauth & Permissions** tab, the _Redirect URL_ should be the one that generated by ngrok.
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/README.md]: Remove the extra `that`
```suggestion
a. In the **Oauth & Permissions** tab, the _Redirect URL_ should be the one generated by ngrok.
```

File: samples/csharp_dotnetcore/53.SlackAdapterBot/README.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+b. In the **Event Subscriptions** tab, fill in the _Request URL_ with the ngrok endpoint adding '/api/messages'.
+
+c. In order to verify these URLs, the bot must to be running with a valid _SlackVerificationToken_ and _SlackClientSigningSecret_.
+
+3 - Back in the bot project, set the credentials (obtained in step 2) in _appsettings.json_.
+
+  SlackVerificationToken
+  SlackBotToken
+  SlackClientSigningSecret
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/README.md]: These three lines are shown as one. Please fix the indentation.

File: samples/csharp_dotnetcore/53.SlackAdapterBot/wwwroot/default.htm [hunk 7]
```diff
@@ -0,0 +1,420 @@
+﻿<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Microsoft.Bot.Sample.Slack</title>
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/wwwroot/default.htm]: Please change to SlackAdapterBot

File: samples/csharp_dotnetcore/53.SlackAdapterBot/wwwroot/default.htm [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+            }
+        };
+    </script>
+</head>
+
+<body>
+    <header class="header">
+        <div class="header-inner-container">
+            <div class="header-icon" style="display: inline-block"></div>
+            <div class="header-text" style="display: inline-block">Microsoft.Bot.Sample.Slack Bot</div>
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/53.SlackAdapterBot/wwwroot/default.htm]: Please change to `SlackAdapterBot`

File: samples/csharp_dotnetcore/csharp_dotnetcore.sln [hunk 9]
```diff
@@ -57,6 +57,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Inspection", "47.inspection
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "QnAMakerActiveLearningBot", "48.qnamaker-active-learning-bot\QnAMakerActiveLearningBot.csproj", "{0D10F464-E767-4F65-BE97-3AFBD886BCF3}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SlackAdapterBot", "53.SlackAdapterBot\SlackAdapterBot.csproj", "{5CEEF2B7-F0EF-4BCD-B3DD-CE449ABCB85C}"
```
Comment (ceciliaavila) [inline samples/csharp_dotnetcore/csharp_dotnetcore.sln]: Please renum the folder to 54 (Webex is the 53)

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by matiasroldan6 — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Pull request closed without merging**